### ISSUE TITLE
vdoprepareforlvm Throw out of range error

### ISSUE THREAD

**Issue by wloot** [dm-vdo/vdo#55]
Hello, I have vdoprepareforlvm failed when upgrading to rhel 9. I found out it's the following command caused by checking logs.
```
# /usr/libexec/vdoprepareforlvm --check /dev/sda1
vdoprepareforlvm: Unexpected error accessing '/dev/sda1' : VDO Status: Out of range
```
where `/dev/sda1` is a BIOS boot partition for booting with legacy bios on GPT disk, with 1mb space.

**Comment by jshimkus-rh** [comment]
> where `/dev/sda1` is a BIOS boot partition for booting with legacy bios on GPT disk, with 1mb space.

For safety any device that is categorized, based on the upgrade scanning, nor more specifically than as a block device is checked for being a VDO.

As part of conversion a VDO has its metadata shifted.  In order to check whether the device is a VDO and its state as to pre- or post-conversion `vdoprepareforlvm --check` attempts to read both possible locations for the metadata.  Given that the shift is 1 MB the error seen is almost certainly because it was not considered that such a small device would be encountered.

**Comment by wloot** [comment]
Yes, I can expect it. Perhaps skiping devices with a specific uuid when checking could be a workaround, in this case `21686148-6449-6E6F-744E-65656445464` for the bios boot partition.

**Comment by lorelei-sakai** [comment]
vdo prepareforlvm has been updated so that it does not check devices that are so small they cannot realistically be vdo devices.
